### PR TITLE
Intermittents

### LINKED ISSUE #12435: Add a build check that fixed intermittents stay fixed
We might want to add something like an `etc/ci/check-fixed-intermittents.sh` that makes sure that any fixed intermittents in #12424 stay fixed. Something like:

```
./mach test-wpt --repeat 100 --release <list of fixed intermittents>
```

cc @aneeshusa @ConnorGBrewster

### PR DESCRIPTION
**Pull request by notriddle**
We probably want to have Buildbot run that script.

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #12435, #11574, and #11561
- [X] These changes do not require tests because reasons

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12629)

### CODE REVIEW

File: etc/ci/check_intermittents.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+REPEAT_COUNT=100
+
+for test_type in wpt css; do
+    while read test_name; do
+        echo "  - Checking $test_name"
```
Comment (Wafflespeanut) [inline etc/ci/check_intermittents.sh]: Maybe a better message? like, `"Checking $test_name because it's a known intermittent..."`

**Comment by Wafflespeanut** [comment]
This is a great idea! :+1:

File: etc/ci/check_intermittents.sh (hunk 1 above)
Comment (notriddle) [inline etc/ci/check_intermittents.sh]: It's inside a script called check_intermittents, and every CI system in the world shows which command was run before its actual stdout. The only reason I included `  - Checking` was to help distinguish it from the output of wptrunner itself.

File: etc/ci/buildbot_steps.yml [hunk 2]
```diff
@@ -27,6 +27,14 @@ mac-nightly:
   - ./mach package --release
   - ./etc/ci/upload_nightly.sh mac
 
+linux-intermittent:
```
Comment (aneeshusa) [inline etc/ci/buildbot_steps.yml]: CI doesn't read from this file yet, it's currently reading from `buildbot/master/files/config/steps.yml` file in the saltfs repo, which is where we'll need to make this change.

Also, let's name these `linux-rel-intermittent` and `mac-rel-intermittent` for consistency.

File: etc/ci/check_intermittents.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
```
Comment (aneeshusa) [inline etc/ci/check_intermittents.sh]: Shebang should be `#!/usr/bin/env bash` per the [style guide](https://github.com/servo/servo/blob/master/docs/STYLE_GUIDE.md).

Comment (aneeshusa) [inline etc/ci/check_intermittents.sh]: Also, please add the `errexit`, `nounset`, and `pipefail` options.

File: etc/ci/check_intermittents.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+REPEAT_COUNT=100
+
+for test_type in wpt css; do
+    while read test_name; do
+        echo "  - Checking $test_name"
+        ./mach test-${test_type} --release --repeat $REPEAT_COUNT "$test_name" > tmp_log.txt
```
Comment (aneeshusa) [inline etc/ci/check_intermittents.sh]: Let's append to the log file always and have Buildbot handle showing the log (per [this comment](https://github.com/servo/saltfs/pull/453/files#r73552639)). Also, please use the full form and quotes `"${some_variable}"` for the variables (`test_type` and `REPEAT_COUNT`).

File: etc/ci/check_intermittents.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+        if [ $? != 0 ]; then
```
Comment (aneeshusa) [inline etc/ci/check_intermittents.sh]: Let `set -o errexit` handle the short circuit exiting.

File: etc/ci/check_intermittents.sh [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+            cat tmp_log.txt
+            exit $?
+        fi
+    done < etc/ci/former_intermittents_${test_type}.txt
```
Comment (aneeshusa) [inline etc/ci/check_intermittents.sh]: Please quote the filename due to the variable substitution.

File: etc/ci/check_intermittents.sh [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+done
+
```
Comment (aneeshusa) [inline etc/ci/check_intermittents.sh]: nit: remove the empty extra newline.

File: etc/ci/check_intermittents.sh (hunk 3 above)
Comment (notriddle) [inline etc/ci/check_intermittents.sh]: Done.

File: etc/ci/check_intermittents.sh [hunk 8]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+REPEAT_COUNT=100
```
Comment (aneeshusa) [inline etc/ci/check_intermittents.sh]: Please put this after we set the bash options (e.g. `errexit`).

**Comment by aneeshusa** [comment]
LGTM aside from the one nit, please also rebase on a new master (to pull in some new shell lints) and squash the fixups.

**Comment by notriddle** [comment]
Rebased and moved.

**Comment by aneeshusa** [comment]
Please fix the tidy failures.

**Pull request merged**